### PR TITLE
Feature: Adding metadata in response

### LINKED ISSUE #1322: Supporting `metadata` in API repsonse
**Is your feature request related to a problem? Please describe.**
We need a metadata field for custom responses for the API

**Describe the solution you'd like**
Adding a Metadata field to the response.Response with `omitempty` tag to remove it when not present.

### PR DESCRIPTION
**Pull request by coolwednesday**
## Pull Request Template

**Description:**
-   Closes #1322 

Screenshot 2024-12-20 at 4 43 11 PM

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

**Comment by Umang01-hash** [comment]
Please also update the documentation and add tests.

**Comment by ccoVeille** [comment]
Adding tests to control make sure how errors are reported would be appreciated

File: pkg/gofr/http/responder_test.go [hunk 1]
```diff
@@ -6,30 +6,77 @@ import (
...
+			contentType:  "application/json",
+			expectedBody: []byte(`{"metadata":{"environment":"stage"},"data":"Hello World from new Server"}`),
+		},
+		{
+			desc: "gofr response type without metadata",
+			data: resTypes.Response{
+				Data: "Hello World from new Server",
+			},
+			contentType:  "application/json",
+			expectedBody: []byte(`{"metadata":null,"data":"Hello World from new Server"}`),
```
Comment (ccoVeille) [inline pkg/gofr/http/responder_test.go:59]: This one is strange. I mean I feel like everyone will have an empty metadata field now. It's not what is expected

Why having metadata:null when there is no metadata ? I mean there is a omitempty tag

File: pkg/gofr/http/responder_test.go [hunk 2]
```diff
@@ -6,30 +6,77 @@ import (
...
+		contentType := recorder.Header().Get("Content-Type")
+		assert.Equal(t, tc.contentType, contentType, "TEST[%d] Failed: %s", i, tc.desc)
+
+		responseBody := recorder.Body.Bytes()
+
+		expected := bytes.TrimSpace(tc.expectedBody)
+
+		actual := bytes.TrimSpace(responseBody)
+
+		assert.Equal(t, expected, actual, "TEST[%d] Failed: %s", i, tc.desc)
```
Comment (ccoVeille) [inline pkg/gofr/http/responder_test.go:79]: assert.JSONEq provides better error messages when it fails

File: pkg/gofr/http/responder_test.go (hunk 1 above)
Comment (coolwednesday) [reply pkg/gofr/http/responder_test.go]: Solved.

File: pkg/gofr/http/responder_test.go (hunk 2 above)
Comment (coolwednesday) [reply pkg/gofr/http/responder_test.go:79]: I think the current test assertions suffice.

**Review by ccoVeille — APPROVED** [review]

**Review by Umang01-hash — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -1,7 +1,8 @@
 module gofr.dev
 
-go 1.22
-toolchain go1.22.10
```
Comment (vipul-rawat) [inline go.mod:4]: Remove all toolchains

**Review by vipul-rawat — APPROVED** [review]

**Pull request merged**